src/expandableCalendar/__test__/index.spec.ts

```diff
@@ -0,0 +1,294 @@
+
+const dashedToday = toMarkingFormat(today);
+let onDateChanged = jest.fn();
+let onMonthChange = jest.fn();
+
+const TestCase = ({
+  expandableCalendarProps,
+  calendarContextProps
+}: {
+  expandableCalendarProps?: Partial<ExpandableCalendarProps>;
+  calendarContextProps?: Partial<CalendarContextProviderProps>;
+} = {}) => {
+  return generateExpandableCalendarWithContext({
+    expandableCalendarProps,
+    calendarContextProps: {
+      date: dashedToday,
+      onMonthChange,
+      onDateChanged,
+      ...calendarContextProps,
+    }
+  });
+};
+
+let driver = new ExpandableCalendarDriver(testIdExpandable, TestCase());
+
+describe('ExpandableCalendar', () => {
+  beforeEach(() => {
+    onDateChanged = jest.fn();
+    onMonthChange = jest.fn();
+    driver = new ExpandableCalendarDriver(testIdExpandable, TestCase({
+      calendarContextProps: {
+        onDateChanged,
+        onMonthChange,
+      },
+    }));
+    jest.useFakeTimers();
+  });
+
+  describe('Props', () => {
+    describe('initialPosition', () => {
+      const driver = new ExpandableCalendarDriver(testIdExpandable, TestCase({expandableCalendarProps: {initialPosition: Positions.OPEN}, calendarContextProps: {onDateChanged, onMonthChange}}));
+      driver.render();
+
+      it('should be expanded if initialPosition prop = "open"', () => {
+        expect(driver.isCalendarExpanded()).toBe(true);
+      });
+    });
+
+    describe('hideKnob', () => {
+      const driver = new ExpandableCalendarDriver(testIdExpandable, TestCase({expandableCalendarProps: {hideKnob: true}}));
+
+      beforeEach(() => {
+        jest.useFakeTimers();
+        driver.render();
+      });
+
+      it('should hide Knob when hideKnob props = true', () => {
+        expect(driver.getKnob()).toBeNull();
+      });
+    });
+  });
+
+  describe('Init', () => {
+
+    beforeEach(() => {
+      driver.render();
+    });
+
+    it("should display today's date", () => {
+      // CalendarList
+      expect(driver.getCalendarList()).toBeDefined();
+      expect(getMonthTitle(driver.getCalendarList().props.data[50])).toBe(getMonthTitle(today));
+
+      // WeekCalendar
+      expect(driver.getWeekCalendar()).toBeDefined();
+      const weekCalendarProps = driver.getWeekCalendar().props;
+      const initialScrollIndex = weekCalendarProps.initialScrollIndex;
+      const currentWeek = weekCalendarProps.data[initialScrollIndex];
+      expect(currentWeek).toBe(dashedToday);
+
+      // container
+      expect(driver.isCalendarExpanded()).toBe(false);
+
+      // events
+      expect(onMonthChange).not.toHaveBeenCalled(); // not called anyways...
+    });
+  });
+
+  describe('Knob', () => {
+    beforeEach(() => {
+      driver.render();
+    });
+
+    it('should expand expandable header ', () => {
+      driver.toggleKnob();
+      jest.runAllTimers();
+
+      expect(driver.isCalendarExpanded()).toBe(true);
+    });
+
+    it('should day press close expandable header', () => {
```

Comment:
I would add another test to check if 'closeOnDayPress' disables this feature

Reply:
added